src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1032,7 +1033,8 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
 
                             if (analyzedInitializers != null)
                             {
-                                // Eventually we will instrument field initializers, but for now we skip them.
+                                // For dynamic analysis, field initializers are instrumented as part of constructors,
+                                // and so are never instrumented here.  
```

Comment:
No, instrumentForDynamicAnalysis might well be true here. If it were sure to be false, then it could be passed as the argument value rather than introducing a new value here.